Feature/data-landing

noahmanger opened the pull request:
Updates the data landing page template based on the new designs. Includes:
- New `GroupedBarChart` component 
- A mock data function to add the top-level totals from the server 

![image](https://cloud.githubusercontent.com/assets/1696495/15980752/5ecc6c64-2f33-11e6-899c-1a6f00c817df.png)
# Remaining to do:
- [x] Finish tooltips
- [x] Add pattern background for in-progress bars
- [x] Update the search area based on the new designs
- [x] Add the top-5 lists
- [ ] Add links to filings from the totals

Depends on https://github.com/18F/fec-style/pull/392

Commit:
Gutting & renaming landing page

Commit:
More landing page code

- Deleting old search.html template

Commit:
New landing page

Commit:
Moving zero pad to helpers

Commit:
Styling charts

Commit:
Adding top lists

Commit:
Adding breakpoint helpers

Commit:
Adding top candidate data

Commit:
Adding stripe bg

Commit:
fixing linter errors

Commit:
Adding links and aria attributes

Commit:
Adding glossary terms

Commit:
Adjusting zero pad styling

Commit:
Updating icons and classes

noahmanger commented:
Update on this:

## What's done:
I have the hero, search area, links to advanced and election pages done:
![image](https://cloud.githubusercontent.com/assets/1696495/16124683/ca0bca22-33a4-11e6-925b-36cc0ef4da44.png)

The raising overview is done:
![image](https://cloud.githubusercontent.com/assets/1696495/16124692/d40c5014-33a4-11e6-82de-534c425cbea1.png)

As is the spending overview:
![image](https://cloud.githubusercontent.com/assets/1696495/16124699/dc05d8b2-33a4-11e6-8316-d22a9c43db7f.png)

## What's not done
- Need "about this data" content (new issue here: https://github.com/18F/openFEC-web-app/issues/1275)
- Need to complete the interaction of the feedback widget (https://github.com/18F/openFEC-web-app/issues/1249)
- The top candidate, pac and party committee lists are incomplete. I have the markup, interactivity and styles coded up, but we're lacking the necessary data, so I've removed it from this branch for now

Those three things can be dealt with separately after merging this in.

Commit:
Removing the top lists

adborden commented:
Has @jenniferthibault reviewed?

adborden commented on openfecwebapp/templates/landing.html:
```diff
@@ -0,0 +1,110 @@
+        </div>
+        <div class="card__content">
+          <h2 class="card__title"><a href="{{ url_for('election_lookup') }}">Elections by location &raquo;</a></h2>
+          <span>Find elections near you— search by state or ZIP code.</span>
```
I don't think there is supposed to be space after the dash. cc @emileighoutlaw ?

adborden commented on static/js/modules/bar-charts.js:
```diff
@@ -0,0 +1,269 @@
+  this.element = d3.select(selector);
+  this.data = data;
+
+  this.margin = {top: 0, right: 20, bottom: 50, left: 40};
```
Can these be pulled from the computed styles? e.g. using `jQuery.css`?

noahmanger replied:
Mmm I don't think so. They're the _future_ properties of the `svg` element created below, and are used to calculate various things before that's appended.

adborden commented on static/js/modules/bar-charts.js:
```diff
@@ -0,0 +1,269 @@
+};
+
+GroupedBarChart.prototype.populateTooltip = function(d) {
+  var entityMap = {
```
Not a big deal, but when mapping internal names to UI names, I prefer to use the term "display names" so it's clear that these are meant for display to the user, not meant for other computation. I might consider naming this "entityDisplayNamesMap". I know, verbose :)

adborden commented on static/js/modules/bar-charts.js:
```diff
@@ -0,0 +1,269 @@
+  this.tooltip
+    .style({'display': 'block'})
+    .style({'top': top.toString() + 'px'})
+    .style({'left': left.toString() + 'px'})
```
This is a jquery element, right? Can you use the dictionary assignment:
```
.style({
  'display': 'block',
  'top': top.toString() + 'px',
  //...
})
```

adborden commented on static/js/modules/helpers.js:
```diff
@@ -180,6 +213,9 @@ module.exports = {
   globals: globals,
-  getTimePeriod: getTimePeriod
+  getTimePeriod: getTimePeriod,
+  zeroPad: zeroPad
```
Not a big deal, maybe alphabetize this list so it's easier to spot duplicates and find what you're looking for?

adborden commented:
👍

jenniferthibault commented:
Haven't reviewed yet, plz don't merge. @noahmanger would you prefer to pair, or should I leave any styling change requests here?

Commit:
Updating text and style

per @emileighoutlaw in
https://github.com/18F/openFEC/issues/1656#issuecomment-225319559

Commit:
Swapping order of cards

Can’t find where this request went through, but this is the order form
the mockups

Commit:
Removing top rule

Commit:
h/t @emileighoutlaw in
https://github.com/18F/openFEC/issues/1656#issuecomment-225319559

Commit:
Removing some extra space we don't need here

jenniferthibault commented:
Got a few things mostly in the spacing and font styles here, but also a few copy things from Emileigh's earlier feedback on the [visual design issue](https://github.com/18F/openFEC/issues/1656#issuecomment-225319559).

<img width="1114" alt="screen shot 2016-06-16 at 3 31 01 pm" src="https://cloud.githubusercontent.com/assets/11636908/16130525/930baeb0-33d7-11e6-971a-0f8f40d488f3.png">

The h3 styles aren't perfect headings here for "Raising overview" and "Spending overview", but I can address that later. 

Bigger fix: the `?` icon on the feedback thing isn't showing up
<img width="255" alt="screen shot 2016-06-16 at 3 33 41 pm" src="https://cloud.githubusercontent.com/assets/11636908/16130548/b9e0bc10-33d7-11e6-9465-6316acdde965.png">

emileighoutlaw commented:
Oh nice! You make my words look so great, Jen!

Commit:
Basic implementation of reaction box

Commit:
Changes from feedback

- Fixing em dash
- Alphabetizing helpers
- Changing variable name
- Consolidating style methods

Thanks @adborden !

noahmanger commented:
Ok! Addressed the feedback. Thanks for the notes -- good stuff.

Commit:
Merge branch 'develop' of github.com:18F/openFEC-web-app into feature/reaction-box

Commit:
Building out reaction box

Commit:
Merge branch 'feature/reaction-box' into feature/data-landing

Commit:
Removing top-candidates api call

noahmanger commented:
Update 2: The reaction box now works. If the user just clicks a reaction button, it sends a google analytics custom event. If they fill out the text box, it logs a feedback issue.

![see](http://g.recordit.co/KkWc44VxBh.gif)

jenniferthibault commented:
Noiiiiiice! 🎈 ⚡

adborden commented on static/js/modules/reaction-box.js:
```diff
@@ -3,36 +3,95 @@
+      eventValue: 1
+    };
+    ga('send', gaEventData);
+  }
```
This should really be a helper method that would detect `ga` and do a no-op if it's unavailable (or even queue it if there's a possibility ga will load later).

adborden commented on static/js/modules/reaction-box.js:
```diff
@@ -3,36 +3,95 @@
+  var labelMap = {
+    'informative': 'Great! \n What did you learn?',
+    'confusing': 'We\'re sorry to hear that. What didn\'t make sense?',
+    'not-interested': 'We\'re sorry to hear that. What would you like to see?',
```
Should these be smart quotes?

adborden commented on static/js/modules/reaction-box.js:
```diff
@@ -3,36 +3,95 @@
+      this.location + ' page',
+    feedback: 'How was it? ' + this.reaction + '\n' +
+      'Tell us more: ' + this.$textarea.val(),
+  };
```
I would prefer to do this kind of formatting on the server side. This allows the API to be more flexible if you're just specifying simple data attributes. e.g.
```javascript
{
  "chart_name": this.name,
  "location":  this.location,
  "reaction": this.reaction,
  "feedback": this.$textarea.val()
}
```

Commit:
Fixes based on review

- Moves GA check to a helper function
- Moves reaction box processing to server-side
- Uses smart quotes for apostrophes

Commit:
Fixing response formatting
